examples/ogbn_papers_100m.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import os
+import os.path as osp
+import time
+
+import torch
+import torch.nn.functional as F
+from ogb.nodeproppred import PygNodePropPredDataset
+from torchmetrics import Accuracy
+
+import torch_geometric
+from torch_geometric.loader import NeighborLoader
+from torch_geometric.nn import GCNConv
+from torch_geometric.utils import to_undirected
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--hidden_channels', type=int, default=64)
+parser.add_argument('--lr', type=float, default=0.01)
+parser.add_argument('--epochs', type=int, default=3)
+parser.add_argument('--batch_size', type=int, default=128)
+parser.add_argument('--fan_out', type=int, default=50)
+
+args = parser.parse_args()
+
+if torch.cuda.is_available():
+    device = torch.device('cuda')
+else:
+    device = torch.device('cpu')
+
+dataset = PygNodePropPredDataset(name='ogbn-papers100M')
+split_idx = dataset.get_idx_split()
+data = dataset[0]
+data.y = data.y.reshape(-1)
+print("Data =", data)
+
+
+def pyg_num_work():
+    num_work = None
+    if hasattr(os, "sched_getaffinity"):
+        try:
+            num_work = len(os.sched_getaffinity(0)) / 2
+        except Exception:
+            pass
+    if num_work is None:
+        num_work = os.cpu_count() / 2
+    return int(num_work)
+
+
+class GCN(torch.nn.Module):
+    def __init__(self, in_channels, hidden_channels, out_channels):
+        super().__init__()
+        self.conv1 = GCNConv(in_channels, hidden_channels)
+        self.conv2 = GCNConv(hidden_channels, out_channels)
+
+    def forward(self, x, edge_index, edge_weight=None):
+        x = F.dropout(x, p=0.5, training=self.training)
+        x = self.conv1(x, edge_index, edge_weight).relu()
+        x = F.dropout(x, p=0.5, training=self.training)
+        x = self.conv2(x, edge_index, edge_weight)
+        return x
+
+
+model = GCN(dataset.num_features, args.hidden_channels,
+            dataset.num_classes).to(device)
+optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=0.0005)
+batch_size = args.batch_size
+train_loader = NeighborLoader(data, num_neighbors=[args.fan_out, args.fan_out],
+                              input_nodes=split_idx['train'],
+                              batch_size=batch_size,
+                              num_workers=pyg_num_work())
+eval_loader = NeighborLoader(data, num_neighbors=[args.fan_out, args.fan_out],
+                             input_nodes=split_idx['valid'],
+                             batch_size=batch_size, num_workers=pyg_num_work())
+test_loader = NeighborLoader(data, num_neighbors=[args.fan_out, args.fan_out],
+                             input_nodes=split_idx['test'],
+                             batch_size=batch_size, num_workers=pyg_num_work())
+eval_steps = 100
+acc = Accuracy(task="multiclass", num_classes=dataset.num_classes).to(device)
+for epoch in range(args.epochs):
+    for i, batch in enumerate(train_loader):
+        if i >= 10:
+            start = time.time()
```

Comment:
nit: do we need this?
```suggestion
```

Reply:
yes, when timing we usually need to allow for some warmup iterations to get accurate average iteration times

Reply:
I was thinking that because this is not a microbenchmark, it'd still be reasonable to measure the overall time incl. warmups, but I'm fine with either way :)